filters/include/pcl/filters/functor_filter.h

```diff
@@ -76,22 +77,48 @@ class FunctorFilter : public FilterIndices<PointT> {
    */
   void
   applyFilter(Indices& indices) override
+  {
+    const auto exec = executor::best_fit{};
+    applyFilter(exec, indices);
+  }
+
+  template <typename Executor, typename executor::instance_of_base<Executor, executor::inline_executor, executor::omp_executor> = 0>
+  void
+  applyFilter (const Executor &exec, Indices& indices)
   {
     indices.clear();
-    indices.reserve(input_->size());
+    indices.reserve(indices_->size());
+    std::vector<std::uint8_t> keep(indices_->size(), 0);
+
     if (extract_removed_indices_) {
       removed_indices_->clear();
-      removed_indices_->reserve(input_->size());
+      removed_indices_->reserve(indices_->size());
     }
 
-    for (const auto index : *indices_) {
-      // functor returns true for points that should be selected
-      if (negative_ != functor_(*input_, index)) {
-        indices.push_back(index);
-      }
-      else if (extract_removed_indices_) {
-        removed_indices_->push_back(index);
+    auto cond_filtering = [&](executor::executor_index_t<Executor> index) {
+      pcl::utils::ignore(index);
+      #pragma omp for
+      for (index_t idx = 0; idx < indices_->size(); ++idx) {
```

Comment:
OMP wants signed index. Please don't use `index_t` here

Reply:
Do we need to define an equivalent `sindex_t`, analogous to `uindex_t`?

Reply:
Seems likely. I was using `std::ptrdiff_t` but `sindex_t` appears better appraoch

Reply:
Switched to `std::ptrdiff_t` as of now. We can switch to `sindex_t` in a separate PR, I reckon there will be other places needing this update too,